OpenSim/Common/DataTable.h

```diff
@@ -646,20 +646,21 @@ class DataTable_ : public AbstractDataTable {
     void appendRow(const ETX& indRow, const RowVectorView& depRow) {
         validateRow(_indData.size(), indRow, depRow);
 
+        try {
+            auto& labels =
+                    _dependentsMetaData.getValueArrayForKey("labels");
+            OPENSIM_THROW_IF(static_cast<unsigned>(depRow.ncol()) !=
+                             labels.size(),
+                             IncorrectNumColumns,
+                             labels.size(),
+                             static_cast<size_t>(depRow.ncol()));
```

Comment:
This try catch block could be avoided by checking `hasKey()`. I realize this is how it was before, so could revise in another PR.

Reply:
I actually really want this change, so I've added it to this PR.

Reply:
👍 Nice!